cli/flox/src/commands/install.rs

```diff
@@ -218,20 +233,97 @@ impl Install {
         );
 
         match err {
+            EnvironmentError::Core(CoreEnvironmentError::LockedManifest(
+                LockedManifestError::ResolutionFailed(failures),
+            )) if failures.0.iter().all(|f| {
+                matches!(f, ResolutionFailure::PackageUnavailableOnSomeSystems { .. })
+            }) =>
+            {
+                let mut packages = packages
+                    .iter()
+                    .cloned()
+                    .map(|p| (p.id().to_string(), p))
+                    .collect::<HashMap<_, _>>();
+
+                for failure in &failures.0 {
+                    let ResolutionFailure::PackageUnavailableOnSomeSystems {
+                        catalog_message:
+                            MsgAttrPathNotFoundNotFoundForAllSystems {
+                                install_id,
+                                valid_systems,
+                                ..
+                            },
+                        invalid_systems: _,
+                    } = failure
+                    else {
+                        unreachable!("already checked that these failures are 'package unavailable on some systems'")
+                    };
+
+                    let Some(package_to_install) = packages.get_mut(install_id) else {
+                        warn!(install_id, "resolution failure for non-existent package");
+                        continue;
+                    };
+
+                    let PackageToInstall::Catalog(CatalogPackage { systems, .. }) =
+                        package_to_install
+                    else {
+                        warn!(
+                            install_id,
+                            ?package_to_install,
+                            "resolution failure for non-catalog package"
+                        );
+                        continue;
+                    };
+
+                    *systems = Some(valid_systems.clone());
+                }
+
+                let packages = packages.into_values().collect::<Vec<_>>();
+
+                let install_result = Dialog {
```

Comment:
Should we `message::warning` when this happens? It's good that we're able to resolve it automatically but it seems surprising that a package install was successful but it's not actually available to my environment on the current system:

```
(2) ~/projects/flox/flox on HEAD (a946fa3b) [$]
% ft install bpftrace   
✅ 'bpftrace' installed to environment 'tmp'

(2) ~/projects/flox/flox on HEAD (a946fa3b) [$]
% ft list            
hello: hello (2.12.1)
```

Reply:
hmm, yeah we'd need to encode that somehow in the installation result though so that we can print something else e.g. `✅ 'bpftrace' (aarch64-linux, x86_64-linux) installed to environment 'tmp'`

Reply:
oh i guess we can also just print 

```
⚠️ installing 'bpftrace' for systems 'aarch64-linux, x86_64-linux' only
```

here and keep the original success message as is

Reply:
Yeah, an additional and separate warning seems fine 👍🏻